src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx

```diff
@@ -68,28 +69,51 @@ export const K8SResourceList = ({
 
     const [resourceListLoader, _resourceList, _resourceListDataError, reloadResourceListData] = useAsync(
         () =>
-            abortPreviousRequests(
-                () =>
-                    getK8sResourceList(
+            abortPreviousRequests(async () => {
+                try {
+                    if (selectedResource.gvk.Kind === SIDEBAR_KEYS.nodeGVK.Kind) {
+                        const response = await getNodeList(clusterId, abortControllerRef.current.signal)
+
+                        return parseNodeList(response)
+                    }
+
+                    return await getK8sResourceList(
                         getResourceListPayload(
                             clusterId,
                             selectedNamespace.value.toLowerCase(),
                             selectedResource,
                             filters,
                         ),
                         abortControllerRef.current.signal,
-                    ).catch((err) => {
-                        if (!getIsRequestAborted(err)) {
-                            showError(err)
-                            throw err
-                        }
-                    }),
-                abortControllerRef,
-            ),
+                    )
+                } catch (err) {
+                    if (!getIsRequestAborted(err)) {
+                        showError(err)
+                        throw err
+                    }
+
+                    return null
+                }
+            }, abortControllerRef),
         [selectedResource, clusterId, selectedNamespace, filters],
-        selectedResource && selectedResource.gvk.Kind !== SIDEBAR_KEYS.nodeGVK.Kind,
     )
 
+    const [, nodeK8sVersions] = useAsync(async () => {
```

Comment:
What will happen in case of error?

Reply:
handled. propagating the error to the select picker